### PR TITLE
Use RCPBasicKeyEq to simplify a test

### PR DESCRIPTION
**Pull request by certik**
As suggested by @isuruf.

### CODE REVIEW

File: symengine/tests/ntheory/test_ntheory.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "catch.hpp"
 #include <chrono>
 
+#include <symengine/basic.h>
```
Comment (certik) [inline symengine/tests/ntheory/test_ntheory.cpp]: We don't need this line, I'll fix it.

**Comment by isuruf** [comment]
There's also another occurrence in the same file.

**Comment by certik** [comment]
Good point, fixed.

**Comment by isuruf** [comment]
+1

**Comment by certik** [comment]
We also have `vec_basic_eq`, but I checked that it doesn't work unless it's a vector of Basic:
```
/home/ondrej/repos/symengine/symengine/tests/ntheory/test_ntheory.cpp:453:38: error: invalid initialization of reference of type ‘const vec_basic& {aka const std::vector<Teuchos::RCP<const SymEngine::Basic> >&}’ from expression of type ‘std::vector<Teuchos::RCP<const SymEngine::Integer> >’
     bool same = vec_basic_eq(v, roots);
                                      ^
```
Which is a shame, as it should work, but I guess there are to many templates in this.

**Pull request merged**